PurchasesTests/Networking/BackendTests.swift

```diff
@@ -1887,6 +1892,115 @@ class BackendTests: XCTestCase {
+                    "onemonth_freetrial": [
+                        "expires_date": futureDateString
+                    ]
+                ]
+            ]
+        ]
+
+        let validUpdatedSubscriberResponse: [String: Any] = [
+            "request_date": "2019-08-16T10:30:42Z",
+            "subscriber": [
+                "first_seen": "2019-07-17T00:05:54Z",
+                "original_app_user_id": "UPDATED",
+                "subscriptions": [
+                    "onemonth_freetrial": [
+                        "expires_date": futureDateString
+                    ],
+                    "twomonth_awesome": [
+                        "expires_date": futureDateString
+                    ]
+                ]
+            ]
+        ]
+        let initialCustomerInfoResponse = HTTPResponse(statusCode: 200, response: validSubscriberResponse)
+        let updatedCustomerInfoResponse = HTTPResponse(statusCode: 200, response: validUpdatedSubscriberResponse)
+        let postResponse = HTTPResponse(statusCode: 200, response: validUpdatedSubscriberResponse)
+        httpClient.mock(requestPath: "/receipts", response: postResponse)
+        httpClient.mock(requestPath: "/subscribers/" + userID, response: initialCustomerInfoResponse)
+
+        var originalSubscriberInfo: CustomerInfo?
+        var updatedSubscriberInfo: CustomerInfo?
+        var postSubscriberInfo: CustomerInfo?
+
+        var callOrder: (initialGet: Bool,
+                        postResponse: Bool,
+                        updatedGet: Bool) = (false, false, false)
+        backend.getSubscriberData(appUserID: userID, completion: { (customerInfo, _) in
+            originalSubscriberInfo = customerInfo
+            callOrder.initialGet = true
+
+            self.httpClient.mocks.removeValue(forKey: "/subscribers/\(self.userID)")
+        })
+
+        backend.post(receiptData: receiptData,
+                     appUserID: userID,
+                     isRestore: false,
+                     productData: nil,
+                     presentedOfferingIdentifier: nil,
+                     observerMode: true,
+                     subscriberAttributes: nil,
+                     completion: { (customerInfo, _) in
+            self.httpClient.mock(requestPath: "/subscribers/" + self.userID, response: updatedCustomerInfoResponse)
+            callOrder.postResponse = true
+            postSubscriberInfo = customerInfo
+        })
+
+        backend.getSubscriberData(appUserID: userID, completion: { (newSubscriberInfo, _) in
+            expect(callOrder) == (true, true, false)
+            updatedSubscriberInfo = newSubscriberInfo
+            callOrder.updatedGet = true
+        })
+
+        expect(callOrder).toEventually(equal((true, true, true)))
+        expect(updatedSubscriberInfo).toEventuallyNot(beNil())
+        expect(updatedSubscriberInfo).toEventually(equal(postSubscriberInfo))
+        expect(updatedSubscriberInfo).toEventuallyNot(equal(originalSubscriberInfo))
```

Comment:
Can we add a expect on the total count here like on the previous test?

```suggestion
        expect(callOrder).toEventually(equal((true, true, true)))
        expect(updatedSubscriberInfo).toEventuallyNot(beNil())
        expect(updatedSubscriberInfo).toEventually(equal(postSubscriberInfo))
        expect(updatedSubscriberInfo).toEventuallyNot(equal(originalSubscriberInfo))
        
        expect(self.httpClient.calls.count) == 3
```

Reply:
Good idea, added to both, and checking what actual requests are being made :)